src/main/java/com/uid2/shared/auth/AuthorizableStore.java

```diff
@@ -0,0 +1,75 @@
+package com.uid2.shared.auth;
+
+import com.uid2.shared.secret.KeyHasher;
+
+import java.nio.ByteBuffer;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+
+public class AuthorizableStore<T extends IAuthorizable> {
+    private class AuthorizableStoreSnapshot {
+        private final Map<ByteBuffer, T> authorizables;
+        private final Collection<byte[]> salts;
+
+        public AuthorizableStoreSnapshot(Collection<T> authorizables) {
+            this.authorizables = authorizables.stream()
+                    .filter(a -> a.getKeyHash() != null) // TODO: remove this filter when all keys have hashes
+                    .collect(Collectors.toMap(
+                            a -> ByteBuffer.wrap(Base64.getDecoder().decode(a.getKeyHash())),
+                            a -> a
+                    ));
+            this.salts = authorizables.stream()
+                    .filter(a -> a.getKeySalt() != null) // TODO: remove this filter when all keys have salts
+                    .map(a -> Base64.getDecoder().decode(a.getKeySalt()))
+                    .collect(Collectors.toList());
+        }
+
+        public Map<ByteBuffer, T> getAuthorizables() {
+            return authorizables;
+        }
+
+        public Collection<byte[]> getSalts() {
+            return salts;
+        }
+    }
+
+    private final AtomicReference<AuthorizableStoreSnapshot> authorizables;
+    private final KeyHasher keyHasher;
```

Comment:
`keyHasher` can be a static final variable

Reply:
Yeah possibly, I left it as non static incase we decide to inject it later on, but if you disagree I can make it static

Reply:
One thing we could do instead is take the `keyHasher` in as a parameter instead in the constructor so we can pass the same `keyHasher` instance from our main Verticle code to all the different providers

Reply:
All the Stores will need a keyHasher for almost no reason, I'd rather make it static then

Reply:
I was referring to making it as a parameter only for `RotatingClientKeyProvider`, `RotatingOperatorKeyProvider`, `AdminUserProvider` since those are the classes that would need to make use of the `keyHasher`. We won't have to do it for others

Reply:
I just made it static 😅